homeassistant/components/solarlog/const.py

```diff
@@ -17,83 +30,192 @@
-    "voltage_ac": ["voltageAC", "voltage AC", ELECTRIC_POTENTIAL_VOLT, "mdi:flash"],
-    "voltage_dc": ["voltageDC", "voltage DC", ELECTRIC_POTENTIAL_VOLT, "mdi:flash"],
-    "yield_day": ["yieldDAY", "yield day", ENERGY_KILO_WATT_HOUR, "mdi:solar-power"],
-    "yield_yesterday": [
-        "yieldYESTERDAY",
-        "yield yesterday",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:solar-power",
-    ],
-    "yield_month": [
-        "yieldMONTH",
-        "yield month",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:solar-power",
-    ],
-    "yield_year": ["yieldYEAR", "yield year", ENERGY_KILO_WATT_HOUR, "mdi:solar-power"],
-    "yield_total": [
-        "yieldTOTAL",
-        "yield total",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:solar-power",
-    ],
-    "consumption_ac": ["consumptionAC", "consumption AC", POWER_WATT, "mdi:power-plug"],
-    "consumption_day": [
-        "consumptionDAY",
-        "consumption day",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "consumption_yesterday": [
-        "consumptionYESTERDAY",
-        "consumption yesterday",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "consumption_month": [
-        "consumptionMONTH",
-        "consumption month",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "consumption_year": [
-        "consumptionYEAR",
-        "consumption year",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "consumption_total": [
-        "consumptionTOTAL",
-        "consumption total",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "total_power": ["totalPOWER", "total power", "Wp", "mdi:solar-power"],
-    "alternator_loss": [
-        "alternatorLOSS",
-        "alternator loss",
-        POWER_WATT,
-        "mdi:solar-power",
-    ],
-    "capacity": ["CAPACITY", "capacity", PERCENTAGE, "mdi:solar-power"],
-    "efficiency": [
-        "EFFICIENCY",
-        "efficiency",
-        f"% {POWER_WATT}/{POWER_WATT}p",
-        "mdi:solar-power",
-    ],
-    "power_available": [
-        "powerAVAILABLE",
-        "power available",
-        POWER_WATT,
-        "mdi:solar-power",
-    ],
-    "usage": ["USAGE", "usage", None, "mdi:solar-power"],
-}
+
+@dataclass
+class SolarlogRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    json_key: str
+
+
+@dataclass
+class SolarLogSensorEntityDescription(
+    SensorEntityDescription, SolarlogRequiredKeysMixin
+):
+    """Describes Solarlog sensor entity."""
+
+
+SENSOR_TYPES: tuple[SolarLogSensorEntityDescription, ...] = (
+    SolarLogSensorEntityDescription(
+        key="time",
+        json_key="TIME",
+        name="last update",
+        device_class=DEVICE_CLASS_TIMESTAMP,
+    ),
+    SolarLogSensorEntityDescription(
+        key="power_ac",
+        json_key="powerAC",
+        name="power AC",
+        icon="mdi:solar-power",
+        unit_of_measurement=POWER_WATT,
```

Comment:
Use `native_unit_of_measurement` instead of `unit_of_measurement` in `SolarLogSensorEntityDescription`

Reply:
Fixed